app/controllers/donations_controller.rb

```diff
@@ -29,4 +27,22 @@ def candidate_id
     def issue_id
       params[:donation][:issue_id]
     end
+
+    def create_cart
+      cart.create(params)
+    end
+
+    def validate_params
+      params[:donation][:amount].to_i > 0
+    end
+
+    def validate_session
+      if session[:donations]
+        session[:donations].merge(create_cart) do |key, old_val, new_val|
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.